src/Datastore/Query/GqlQuery.php

```diff
@@ -0,0 +1,245 @@
+ * parameter binding is highly recommended.
+ *
+ * Idiomatic usage is via {@see Google\Cloud\Datastore\DatastoreClient::gqlQuery()}.
+ *
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\ServiceBuilder;
+ *
+ * $cloud = new ServiceBuilder();
+ *
+ * $datastore = $cloud->datastore();
+ *
+ * $query = $datastore->gqlQuery('SELECT * FROM Companies WHERE companyName = @companyName', [
+ *     'bindings' => [
+ *         'companyName' => 'Google'
+ *     ]
+ * ]);
+ *
+ * $res = $datastore->runQuery($query);
+ * foreach ($res as $company) {
+ *     echo $company['companyName']; // Google
+ * }
+ * ```
+ *
+ * ```
+ * // Positional binding is also supported
+ * $query = $datastore->gqlQuery('SELECT * FROM Companies WHERE companyName = @1', [
+ *     'bindings' => [
+ *         'Google'
+ *     ]
+ * ]);
+ * ```
+ *
+ * ```
+ * // While not recommended, you can use literals in your query string:
+ * $query = $datastore->gqlQuery("SELECT * FROM Companies WHERE companyName = 'Google'", [
+ *     'allowLiterals' => true
+ * ]);
+ * ```
+ *
+ * @see https://cloud.google.com/datastore/docs/apis/gql/gql_reference GQL Reference
+ */
+class GqlQuery implements QueryInterface
+{
+    use DatastoreTrait;
+
+    const BINDING_NAMED = 'namedBindings';
+    const BINDING_POSITIONAL = 'positionalBindings';
+
+    /**
+     * @var string
+     */
+    private $query;
+
+    /**
+     * @var string
+     */
+    private $projectId;
+
+    /**
+     * @var array
+     */
+    private $options;
+
+    /**
+     * @var array
+     */
+    private $allowedBindingTypes = [
+        self::BINDING_NAMED,
+        self::BINDING_POSITIONAL,
+    ];
+
+    /**
+     * @param string $query The GQL Query string.
+     * @param string $projectId Your Google Cloud Platform Project ID.
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type bool $allowLiterals Whether literal values will be allowed in
+     *           the query string. This is false by default, and parameter
+     *           binding is strongly encouraged over literals.
+     *     @type array $bindings An array of values to bind to the query string.
+     *           Queries using Named Bindings should provide a key/value set,
+     *           while queries using Positional Bindings must provide a simple
+     *           array.
+     *           Applications with no need for multitenancy should not set this value.
+     *     @type string $readConsistency If not using a {@see Google\Cloud\Datastore\Transaction},
+     *           $readConsistency can be set to `STRONG` or `EVENTUAL`.
+     *     @type Transaction $transaction If an instance of {@see Google\Cloud\Datastore\Transaction}
+     *           is given, query will run in the transaction.
+     * }
+     */
+    public function __construct($query, $projectId, array $options = [])
+    {
+        $this->query = $query;
+        $this->projectId = $projectId;
```

Comment:
Do `GqlQuery` and `Query` need the `projectId`? I don't see them in use in either class.